repository/SmalltalkCI-Coverage-Core.package/SCICodeCoverage.class/instance/coverageReportUploadActionScript.st

```diff
@@ -0,0 +1,11 @@
+accessing
+coverageReportUploadActionScript
+
+	"This script is sourced in 'upload-coverage.sh script and executed after the image is closed."
+	self uploadsCoverageReportAutomatically
+		ifTrue: [ 
+			self serviceName = #coveralls
+				ifTrue: [ ^ 'upload-to-coveralls.sh' ].
+			self serviceName = #codecov
+				ifTrue: [ ^ 'upload-to-codecov.sh' ] ].
+	^ 'skip-coverage-upload.sh'
```

Comment:
Mh, now that I see this, I don't like the fact that the Smalltalk knows about bash script names. Could we rename `upload-coverage.sh` to something like `coverage.env` and store `readonly COVERAGE_SERVICE="{serviceName}"` in there? On the bash side, we can then source `coverage.env` and check `$COVERAGE_SERVICE` for its contents. What do you think?

Reply:
I think it's a good idea. It was the first thing I've proposed: using some env variables. I have some busy weeks ahead so if you want to proceed with the proposed changes go ahead.